packages/sync-rules/src/BucketSource.ts

```diff
@@ -69,13 +83,153 @@ export interface BucketSource {
   resolveResultSets(schema: SourceSchema, tables: Record<string, Record<string, ColumnDefinition>>): void;
 
   debugWriteOutputTables(result: Record<string, { query: string }[]>): void;
+}
 
-  debugRepresentation(): any;
+/**
+ * A parameter lookup source defines how to extract parameter lookup values from parameter queries.
+ *
+ * This is only relevant for parameter queries that query tables.
+ */
+export interface BucketParameterLookupSourceDefinition {
```

Comment:
nit: Maybe it's just me, but these names don't sound particularly descriptive or helpful to me (and I keep getting `BucketParameterLookupSourceDefinition` and `BucketParameterQuerierSourceDefinition` mixed up).

Since sync streams already make the definition of parameters a bit blurry, I wonder if other terms could be more intuitive here:

- For `BucketParameterQuerierSource`, maybe something like `ConnectionBucketsQuerier` or `ConnectionBucketsResolver` makes it clearer what this is actually doing.
- For `BucketParameterLookupSource`, maybe something like `IndexLookupCreator` since that's really what parameter lookups are? Then if we need different kind of indexes in the future (say a range index for IVM with order by), we could potentially add those more easily.

Reply:
I went with ParameterIndexLookupCreator here now. I still want to keep the "parameter" part: Even though "parameter query" not a user-facing concept in sync streams, the name is still used in the implementation in many places, and the concept still translates to sync stream subqueries.

Reply:
I removed the concept of BucketParameterQuerierSourceDefinition now. These don't persist any data, and we never need to evaluate them individually, so it made more sense to have the functionality on HydratedBucketSource.

I kept the implementation for these queriers mostly the same, but the structure is an implementation detail now, rather than a public API.